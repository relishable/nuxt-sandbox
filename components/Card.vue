<template>
  <div class="column">
    <div class="card">
      <header class="card-header">
        <p class="card-header-title has-text-grey">
          {{ state.title }}
        </p>
      </header>
      <div class="card-content">
        <div class="content has-text-centered">
          <b-icon :icon="state.icon" size="is-large" type="is-primary" />
        </div>
      </div>
      <footer class="card-footer">
        <div class="card-footer-item">
          <span>
            <slot />
          </span>
        </div>
      </footer>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, reactive } from '@nuxtjs/composition-api'

export default defineComponent({
  props: {
    title: {
      type: String,
      required: true,
    },
    icon: {
      type: String,
      required: true,
    },
  },
  setup(props) {
    const state = reactive({
      title: computed(() => props.title),
      icon: computed(() => props.icon),
    })

    return { state }
  },
})
</script>
