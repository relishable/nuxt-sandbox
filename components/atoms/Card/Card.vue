<template>
  <div class="column">
    <div class="card">
      <header class="card-header">
        <p class="card-header-title has-text-grey">
          {{ state.card.title }}
        </p>
      </header>
      <div class="card-content">
        <div class="content has-text-centered">
          <b-icon :icon="state.card.icon" size="is-large" type="is-primary" />
        </div>
      </div>
      <footer class="card-footer">
        <div class="card-footer-item">
          <span v-html="state.card.content"></span>
        </div>
      </footer>
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, reactive } from '@nuxtjs/composition-api'

export default defineComponent({
  props: {
    title: {
      type: String,
      required: true,
    },
    icon: {
      type: String,
      required: true,
    },
    content: {
      type: String,
      required: true,
    },
  },
  setup(props) {
    const state = reactive({
      card: computed(() => props),
    })

    return { state }
  },
})
</script>
