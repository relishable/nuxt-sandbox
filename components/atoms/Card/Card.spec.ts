import { Wrapper, shallowMount, mount } from '@vue/test-utils'
import Card from '~/components/atoms/Card/Card.vue'

describe('Card', () => {
  let wrapper: Wrapper<any>
  let shallowWrapper: Wrapper<any>

  beforeEach(() => {
    wrapper = mount(Card, {
      propsData: {
        title: 'Test',
        icon: 'github',
        content: '<a href="https://github.com/">Github</a>',
      },
    })
    shallowWrapper = shallowMount(Card, {
      propsData: {
        title: 'Test',
        icon: 'github',
        content: '<a href="https://github.com/">Github</a>',
      },
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(shallowWrapper.vm).toBeTruthy()
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
