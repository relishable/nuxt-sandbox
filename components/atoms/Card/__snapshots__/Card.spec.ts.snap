// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card initialized snapshot 1`] = `
<div
  class="column"
>
  <div
    class="card"
  >
    <header
      class="card-header"
    >
      <p
        class="card-header-title has-text-grey"
      >
        
        Test
      
      </p>
    </header>
     
    <div
      class="card-content"
    >
      <div
        class="content has-text-centered"
      >
        <b-icon-stub
          icon="github"
          size="is-large"
          type="is-primary"
        />
      </div>
    </div>
     
    <footer
      class="card-footer"
    >
      <div
        class="card-footer-item"
      >
        <span>
          <a
            href="https://github.com/"
          >
            Github
          </a>
        </span>
      </div>
    </footer>
  </div>
</div>
`;
